src/plugins/intel_gpu/src/kernel_selector/kernels/gemm/gemm_kernel_tiled_opt.cpp

```diff
@@ -205,14 +205,10 @@ JitConstants GemmKernelTiledOpt::GetJitConstants(const gemm_params& params) cons
         else
             jit.AddConstant(MakeJitConstant("TRANSPOSE_OUTPUT", 0 /* set as TRANSPOSE_X_LAST */));
 
-        bool has_dynamic_k_padding = params.transpose_input0 ? params.inputs[0].Y().pad.is_dynamic
-                                                             : params.inputs[0].X().pad.is_dynamic;
-        bool has_dynamic_n_padding = params.transpose_input1 ? params.inputs[1].Y().pad.is_dynamic
-                                                             : params.inputs[1].X().pad.is_dynamic;
+        bool has_dynamic_k_padding = params.transpose_input0 ? dims0_padded.dims_padded[input0_dims[6]]
```

Comment:
why should we fix this & jitter?

Reply:
inputs[0].X() is not actually used x value in kernel because the order is transposed. So this is to find transposed actual x and y.

Reply:
1. if the value is not correct due to the transpose, why not checking transposed order? 
2. with your current change in jitter, dim_pad is only stands for "dynamic padding", also it is true when the dim is dynamic even though the pading dim is not dynamic .... which is weird.

Reply:
I've updated define whether each input has dynamic padding on any position because the offset value in the kernel is for padding of all position.